icu4c/source/common/unicode/platform.h

```diff
@@ -206,6 +206,17 @@
 #   define U_PLATFORM U_PF_UNKNOWN
 #endif
 
+/**
+ * \def U_REAL_MSVC
+ * Defined if the compiler is the real MSVC compiler (and not something like
+ * Clang setting _MSC_VER in order to compile Windows code that requires it).
+ * Otherwise undefined.
+ * @internal
+ */
+#if (defined(_MSC_VER) && !(defined(__clang__) && __clang__)) || defined(U_IN_DOXYGEN)
```

Comment:
Sorry, I had some days off last week.

Shouldn't we use `_LIBCPP_VERSION` instead of `__clang__`, as recommended by our colleague working in this area? \
This seems to be more of a difference in libraries, not in the compiler.

Reply:
`_LIBCPP_VERSION` identifies LLVM's libc++, which often is used together with the Clang compiler but isn't really tied to any compiler. The expression here attempts to identify the compiler, regardless of which implementation of the standard library is being used.